utils/csv2img-exif

```diff
@@ -0,0 +1,179 @@
+# [1] = file path, relative to the repository root
+# [2] = license name(s) -> Copyright Exif tag
+# [3] = authorship information -> Artist Exif tag
+# [4] = notes -> UserComment Exif tag
+# [5] = new git commit date (not used) -> ModifyDate (today)
+# [6] = md5 hash (not used, not updated)
+##
+
+##
+# Add new licenses to this list:
+# Avoid things like "GNU GPL v2+;CC BY-SA 4.0", unless you mean to dual license
+# under either GNU GPL v2+ or CC BY-SA 4.0.  GNU GPL v2+ and CC BY-SA 4.0 (e.g.
+# a GNU GPL v2+ file with CC BY-SA 4.0 modifications) isn't legally possible.
+##
+known_licenses = (
+        "CC BY-SA 4.0",
+        "CC0",
+        "GNU GPL v2+",
+        )
+# to check that the date has a four-digit pattern (the year), meaning it really is a date
+dv = re.compile('\d\d\d\d/')
+
+# "-all=" to clear out existing metadata before writing the data from the CSV
+def write_exif(target,date_value,copyright_value,artist_value,usercomment_value):
+    if dv.match(date_value) != None:
+        date_value = datetime.strptime(str(date_value),"%Y/%m/%d").strftime("%Y:%m:%d %H:%M:%S")
+    else:
+        date_value = datetime.today().strftime("%Y:%m:%d %H:%M:%S")
+    return str(check_output(["exiftool", "-overwrite_original", "-all=", "-ModifyDate=now", "-CreateDate=%s" % date_value, "-Copyright=%s" % copyright_value, "-Artist=%s" % artist_value, "-UserComment=%s" % usercomment_value, target]), 'UTF-8')
+
+def fix_webp(target):
+    return str(check_output(["cwebp", "-quiet", "mt", "-z", "9", "-metadata", "exif", target, "-o", target]), 'UTF-8')
+
+
+##
+# program logic start
+##
+
+args = argparse.ArgumentParser()
+args.add_argument("--repo", default="..", help="The directory of the Wesnoth repository to run this script against.")
+args.add_argument("--input", default="copyrights.csv", help="The file to read the existing copyright data from.")
+args.add_argument("--output", default="csv2img-exif_output.csv", help="The file to write the results of this script to.")
+options = args.parse_args()
+os.chdir(options.repo)
+
+
+csv_data = {}
+# Too few fields
+missing_fields = []
+# Too many fields, possibly due to an unquoted comma
+extra_fields = []
+# previously untracked images
+added = []
+# lacking something in either the license or author fields
+incomplete = []
+unchanged = []
+removed = []
+unknown_licenses = []
+
+
+with open(options.input, encoding="utf-8") as csvfile:
+    reader = csv.reader(csvfile)
+    previous_file = ""
+    for row in reader:
+        if row[0] == "Date":
+            continue
+
+        file = row[1]
+        previous_file = file
+
+        if not os.path.exists(file):
+            removed.append(file)
+            continue
+
+        csv_data[file] = row
+
+with contextlib.suppress(FileNotFoundError):
+    os.remove(options.output)
+
+# only transfer the info from complete, properly formatted CSV rows (unchanged)
+# but keep track of all the rest for summary output
+for root, _, files in os.walk(options.repo):
```

Comment:
Since the current directory was already changed to `options.repo` this should just go through the current dir. Otherwise this goes wrong for relative paths.

Reply:
```suggestion
for root, _, files in os.walk("."):
```